service/controller/control_plane.go

```diff
@@ -119,7 +136,14 @@ func newControlPlaneResources(config ControlPlaneConfig) ([]resource.Interface,
 	}
 
 	resources := []resource.Interface{
+		// Following resources manage CR status information. Note that
+		// keepForInfraRefsResource needs to run before
+		// controlPlaneStatusResource because keepForInfraRefsResource keeps
+		// finalizers where contrplPlaneStatusResource does not.
```

Comment:
```suggestion
		// Following resources manage CR status information. Note that
		// keepForInfraRefsResource needs to run before
		// controlPlaneStatusResource because keepForInfraRefsResource keeps
		// finalizers where controlPlaneStatusResource does not.
```

Reply:
I think we should get the design more right in the future so that we do not create such implicit dependencies between handlers. This is the whole reason why we want to get rid of the controller context as well.